src/domain/Reservation/components/ReservationCalendar.tsx

```diff
@@ -0,0 +1,252 @@
+  { id: '7', title: '취소', date: '2025-07-17', color: 'red', number: 7 },
+  { id: '8', title: '예약', date: '2025-07-17', color: 'purple', number: 8 },
+  { id: '9', title: '완료', date: '2025-07-17', color: 'green', number: 9 },
+  { id: '10', title: '취소', date: '2025-07-18', color: 'red', number: 10 },
+];
+
+export default function ReservationCalendar() {
+  const [currentDate, setCurrentDate] = useState(dayjs()); //현재 시간을 가져옴
+  const [events] = useState<Event[]>(sampleEvents); //이벤트 데이터를 가져옴
+
+  // 이벤트 데이터를 날짜별로 그룹화하여 memoization으로 성능 최적화 (샘플 이벤트에 적용되는 내용이라 수정 실제 API 연동 시 수정 필요)
+  const eventsByDate = useMemo(() => {
+    const eventMap: Record<string, Event[]> = {};
+
+    events.forEach((event) => {
+      if (!eventMap[event.date]) {
+        eventMap[event.date] = [];
+      }
+      eventMap[event.date].push(event);
+    });
+    return eventMap;
+  }, [events]);
+
+  // 핵심: dayjs 객체와 Moment.js와 유사한 API를 사용합니다.
+  const monthStart = currentDate.startOf('month'); // 현재 월의 시작 날짜
+  const monthEnd = currentDate.endOf('month'); // 현재 월의 마지막 날짜
+  const calendarStart = monthStart.startOf('week'); // 현재 월의 첫 번째 주의 시작 날짜
+  const calendarEnd = monthEnd.endOf('week'); // 현재 월의 마지막 주의 마지막 날짜
+
+  // 날짜 배열을 생성하기 위해 반복문 사용 (currentDate 기준으로 한 달의 날짜를 가져옴)
+  const days = useMemo(() => {
+    const daysArr = [];
+    let day = calendarStart;
+    while (day.isSameOrBefore(calendarEnd)) {
+      daysArr.push(day);
+      day = day.add(1, 'day');
+    }
+    return daysArr;
+  }, [calendarStart, calendarEnd]);
+
+  /**
+   * 특정 날짜의 이벤트 목록을 반환합니다
+   * @param date - 조회할 날짜 (dayjs 객체)
+   * @returns 해당 날짜의 이벤트 배열
+   */
+  const getEventsForDate = (date: dayjs.Dayjs) => {
+    const formattedDate = date.format('YYYY-MM-DD');
+    return eventsByDate[formattedDate] || [];
+  };
+
+  /**
+   * 이벤트를 우선순위에 따라 정렬합니다
+   * @param events - 정렬할 이벤트 배열
+   * @returns 우선순위 순으로 정렬된 이벤트 배열 (완료 > 승인 > 예약 > 거절 > 취소)
+   */
+  const sortEventsByPriority = (events: Event[]) => {
+    return events.sort((a, b) => {
+      const priorityA =
+        PRIORITY_MAP[a.title as keyof typeof PRIORITY_MAP] ?? 999;
+      const priorityB =
+        PRIORITY_MAP[b.title as keyof typeof PRIORITY_MAP] ?? 999;
+      return priorityA - priorityB;
+    });
+  };
+
+  /**
+   * 이벤트 색상에 따른 Tailwind CSS 클래스를 반환합니다
+   * @param color - 이벤트 색상 ('red' | 'blue' | 'orange' | 'green' | 'purple')
+   * @returns Tailwind CSS 클래스 문자열
+   */
+  const getColorClass = (color: string) => {
+    const colorMap = {
+      red: 'bg-red-200 text-red-400',
+      blue: 'bg-blue-200 text-blue-400',
+      orange: 'bg-orange-200 text-orange-400',
+      green: 'bg-green-200 text-green-400',
+      purple: 'bg-purple-200 text-purple-400',
+    };
+    return colorMap[color as keyof typeof colorMap] || 'bg-gray-500';
+  };
+
+  const prevMonth = () => setCurrentDate(currentDate.subtract(1, 'month'));
+  const nextMonth = () => setCurrentDate(currentDate.add(1, 'month'));
```

Comment:
`currentDate`가 바뀔 때마다 `prevMonth`, `nextMonth`새로 생성됩니다. `useCallback`을 사용해 최적화하는 방법도 고려해보세요!

```suggestion
  const prevMonth = useCallback(() => {
    setCurrentDate((prev) => prev.subtract(1, 'month'));
  }, []); // 종속성 배열이 비어있으면 setCurrentDate가 안정적이라 괜찮습니다.

  const nextMonth = useCallback(() => {
    setCurrentDate((prev) => prev.add(1, 'month'));
  }, []);
```